app/models/core.ts

```diff
@@ -178,7 +185,51 @@ export class MappedPropertyList implements PropertyList {
     get(property: string) {
         return this.backerObject[this.key(property)];
     }
+
     set(property: string, value: any) {
         this.backerObject[this.key(property)] = value;
     }
+
+    hide(mask: Set<string>) {
+        const masked = new Set<string>();
+        for (let i = this.properties.length - 1; i >= 0; i--) {
+            const prop = this.propertyMap.get(this.properties[i][0]);
+            if (prop && mask.has(prop)) {
+                masked.add(prop);
+                this.properties.splice(i, 1);
```

Comment:
Another idea would be to just create a properties array to avoid the potential O(n^2) behavior.